forem/forem
# Add boolean attribute main_image_from_frontmatter to indicate if cove…

## What type of PR is this? (check all applicable)


## Description
Added a new boolean field to articles table. This will enable us to keep track if cover_image/main_image was set from front matter. If so , then if next time we do not receive the field cover_image from front matter, we can clear out the cover image.

## Related Tickets & Documents
Directly related to #15233 

## QA Instructions, Screenshots, Recordings
Altered slightly from the issue reproduction steps
1.  Click on the "Create Post" button
2. Fill in the Jekyll Front Matter (JFM) information for the post (title, published: true, some description, some tag, cover_image). Make sure to set cover_image
3. Save the post (it will be published because JFM published is true)
4. You'll be taken to the published article, now click on the "Edit" button
5. Remove the cover_image line from JFM
6. Save the post (it will be automatically republished because published: true)
7.  Cover image is now removed, earlier  it was not

## Added/updated tests?



## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
Migrate the database
